### PR TITLE
Search in note

### PR DESCRIPTION
**Pull request by mario**
Super sucky code with lots of duplication, but those two fragments and the way it's currently done make no sense to me anyway :P

It works, and this is the first iteration of in-note search, so please test.

### CODE REVIEW

**Comment by korelstar** [comment]
🚀 👍   (but I didn't test it yet)

> Super sucky code with lots of duplication, but those two fragments and the way it's currently done make no sense to me anyway :P

Why it doesn't make any sense? The idea was to put code which is common for both fragments into the `BaseNoteFragment`. Hence, you could/should move all of your search-code into `BaseNoteFragment`. There is no need for duplicated code. ;-)

**Comment by mario** [comment]
@korelstar ah. Well, there's very similar code in two of the fragments that extend it too, so .. :)

But yea, will move it.

**Comment by mario** [comment]
Ready for test & review.

File: app/src/main/java/it/niedermann/owncloud/notes/android/fragment/NoteEditFragment.java [hunk 1]
```diff
@@ -32,12 +33,41 @@
...
+        @Override
+        public void afterTextChanged(final Editable s) {
+            unsavedEdit = true;
+            if (!saveActive) {
+                handler.removeCallbacks(runAutoSave);
+                handler.postDelayed(runAutoSave, DELAY);
+            }
+        }
+    };
+    private android.support.v7.widget.SearchView searchView;
```
[the remark this answers is not in the document]
Comment (mario) [reply app/src/main/java/it/niedermann/owncloud/notes/android/fragment/NoteEditFragment.java]: Good catch.

File: app/src/main/java/it/niedermann/owncloud/notes/android/fragment/NoteEditFragment.java [hunk 2]
```diff
@@ -66,6 +96,10 @@ public void onPrepareOptionsMenu(Menu menu) {
         super.onPrepareOptionsMenu(menu);
         menu.findItem(R.id.menu_edit).setVisible(false);
         menu.findItem(R.id.menu_preview).setVisible(true);
+
+        if (TextUtils.isEmpty(searchQuery)) {
+            searchMenuItem.collapseActionView();
+        }
```
[the remark this answers is not in the document]
Comment (mario) [reply app/src/main/java/it/niedermann/owncloud/notes/android/fragment/NoteEditFragment.java]: Probably not.

**Comment by jancborchardt** [comment]
Screenshots for design review would be very appreciated. :)

**Comment by mario** [comment]
Yes, yes @jancborchardt - as soon as I get the time to update the PR.

**Comment by stefan-niedermann** [comment]
Screenshot as requested by @jancborchardt 

device-2018-06-19-202003

**Comment by jancborchardt** [comment]
How does it work for long notes? - Does it scroll down to the first result automatically?
- Can you move to the next/previous result?
- Are results indicated on the scrollbar? (like in Chrome browser when you Ctrl-F)

**Comment by stefan-niedermann** [comment]
@jancborchardt no, it is basically highlighting. It would be a nice first step, but it has to work at least for [all the cases i mentioned](https://github.com/stefan-niedermann/nextcloud-notes/pull/400#issuecomment-398497294) before i merge.

@mario did you have any time to look up into this PR? :)

**Comment by mario** [comment]
@stefan-niedermann will take a look this weekend. Prev/Next buttons are the future. Step by step.

**Comment by stefan-niedermann** [comment]
> On both, only the first or the first two occurrences of the search string in the note are highlighted (e.g. if I search for test in the note Test test test, the result is "Test test test" and not "Test test test").

Should be fixed in 910e14b37e253ef50f482d980d036922c3a5d4a9

**Comment by stefan-niedermann** [comment]
> On my phone, I have some weird behavior regarding the menu items: By default, the search menu item is shown in the ActionBar. If I start a search and type some chars, the search menu item is not shown anymore (which is intended). However, if I use the three-dot-menu then, the search is still active and the search menu item is shown in the pull down menu. After closing and reopening the three-dot-menu again, the search menu item isn't shown anymore. This behavior is reproducible (Android 6.0.1).

If i use `always` instead of `ifRoom` on both, `search` and `preview`/`edit` the behaviour seems to be correct.

**Comment by stefan-niedermann** [comment]
Okay, let's get this in. The second point of @korelstar is still not resolved, though the benefits are better than the disadvantages. We will tracking this in a separate issue.

**Comment by mario** [comment]
@stefan-niedermann pancakes time?! :D

**Comment by stefan-niedermann** [comment]
@mario are you at the hackweek next month in stuttgart?

**Pull request merged**